controllers/metal3machinetemplate_controller_test.go

```diff
@@ -0,0 +1,214 @@
+			} else if testCase.m3mTemplateIsPaused {
+				fakeManagerFactory.EXPECT().NewMachineTemplateManager(gomock.Any(), gomock.Any(), gomock.Any()).Return(fakeM3MTemplateManager, nil)
+			} else if testCase.common.shouldSucceed {
+				fakeManagerFactory.EXPECT().NewMachineTemplateManager(gomock.Any(), gomock.Any(), gomock.Any()).Return(fakeM3MTemplateManager, nil)
+				fakeM3MTemplateManager.EXPECT().UpdateAutomatedCleaningMode(context.TODO()).Return(nil)
+			}
+
+			result, err := testReconciler.Reconcile(context.TODO(), testCase.common.testRequest)
+			Expect(result).To(Equal(testCase.common.expectedResult))
+			evaluateM3MTemplateTestError(testCase.common.expectedError, err)
+			mockController.Finish()
+		},
+		Entry("Case: M3MTemplate haven't been found",
+			reconcileTemplateTestCase{
+				common: commonTestCase{
+					testRequest:    defaultTestRequest,
+					expectedResult: ctrl.Result{},
+					expectedError:  nil,
+				},
+				m3mTemplateCantBeFound: true,
+			}),
+		Entry("Case: Failed to create helper for managing the template manager",
+			reconcileTemplateTestCase{
+				common: commonTestCase{
+					testRequest:    defaultTestRequest,
+					expectedResult: ctrl.Result{},
+					expectedError:  []byte("failed to create helper for managing the templateMgr"),
+					m3mTemplate:    newMetal3MachineTemplate(clusterName, metal3machineTemplateName, namespaceName, false),
+				},
+				failedToCreateMachineTemplateManager: true,
+			}),
+		Entry("Case: Metal3MachineTemplate is currently paused",
+			reconcileTemplateTestCase{
+				common: commonTestCase{
+					testRequest:    defaultTestRequest,
+					expectedResult: ctrl.Result{Requeue: true, RequeueAfter: requeueAfter},
+					expectedError:  nil,
+					m3mTemplate:    newMetal3MachineTemplate(clusterName, metal3machineTemplateName, namespaceName, true),
+				},
+				m3mTemplateIsPaused: true,
+			}),
+		Entry("Case: Successful Reconcile execution",
+			reconcileTemplateTestCase{
+				common: commonTestCase{
+					testRequest:    defaultTestRequest,
+					expectedResult: ctrl.Result{},
+					expectedError:  nil,
+					m3mTemplate:    newMetal3MachineTemplate(clusterName, metal3machineTemplateName, namespaceName, false),
+					shouldSucceed:  true,
+				},
+			}),
+	)
+
+	DescribeTable("Metal3MachineTemplate reconcileNormal test",
+		func(testCase reconcileTemplateNormalTestCase) {
+			mockController = gomock.NewController(GinkgoT())
+			fakeM3MTemplateManager = baremetal_mocks.NewMockTemplateManagerInterface(mockController)
+			fakeManagerFactory = baremetal_mocks.NewMockManagerFactoryInterface(mockController)
+			fakeClientBuilder = fake.NewClientBuilder()
+			objects = []client.Object{}
+			objects = append(objects, testCase.common.m3mTemplate)
+			fakeClient = fakeClientBuilder.WithScheme(setupScheme()).WithObjects(objects...).Build()
+
+			if testCase.clonedFromNameNotFound {
+				fakeM3MTemplateManager.EXPECT().UpdateAutomatedCleaningMode(context.TODO()).Return(errors.New(""))
```

Comment:
`UpdateAutomatedCleaningMode` can fail for reasons that aren't just `clonedFromNameNotFound` (e.g. https://github.com/metal3-io/cluster-api-provider-metal3/blob/a5917fd0555088a89118927f29cea605766ea8f4/baremetal/metal3machinetemplate_manager.go#L98). The log message in the controller code is misleading. Probably all we can say is that `UpdateAutomatedCleaningMode` failed for some reason and check from the controller that this is handled properly. I would rename `clonedFromNameNotFound` to `updateAutomatedCleaningModeFails`. What do you think?

Reply:
Fixed according to your recommendation.